src/Microsoft.DocAsCode.MarkdownLite/Matchers/MatchContent.cs

```diff
@@ -0,0 +1,72 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.DocAsCode.MarkdownLite.Matchers
+{
+    using System;
+
+    public struct MatchContent
+    {
+        public readonly string Text;
+        public readonly int StartIndex;
+        public readonly bool IsForward;
+
+        public MatchContent(string text, int startIndex, bool isForward = true)
+        {
+            if (text == null)
+            {
+                throw new ArgumentNullException(nameof(text));
+            }
+            if (startIndex < 0 || startIndex > text.Length)
```

Comment:
it can be text.length, and it can match end of string for forward, or begin of string for backward.

---
In reply to: [104572120](https://github.com/dotnet/docfx/pull/1388#discussion_r104572120) [](ancestors = 104572120)